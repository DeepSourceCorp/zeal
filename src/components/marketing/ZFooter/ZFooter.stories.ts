import ZFooter from './ZFooter.vue'
import ZFooterList from '../ZFooterList'
import ZFooterListItem from '../ZFooterListItem'
import ZIcon from '../../ZIcon'

export default {
  title: 'Zeal Marketing/Footer',
  component: ZFooter,
  excludeStories: /.*Data$/,
  parameters: {
    layout: 'fullscreen'
  }
}

export const Basic = () => ({
  components: { ZFooter, ZFooterList, ZFooterListItem, ZIcon },
  template: `<div class='h-screen'>

    <div class="prose prose-md max-w-none p-20">
      <h1 id="this-is-main">Some random heading to grab attention</h1>
      <p class="lead">I dont' have any money, but what I do have is a specific set of skills. Skills that can be nightmare to people like you...</p>
      <h1 id="this-is-main">Some random heading to grab attention</h1>
      <p class="lead">I dont' have any money, but what I do have is a specific set of skills. Skills that can be nightmare to people like you...</p>
    </div>

    <z-footer class="bg-gradient-dawn animate-gradient bg-400%" container="lg">
      <template slot="brand">
        <span class="flex w-20 mb-12 lg:mb-0">
          <img class="max-w-none h-8" src="https://i.imgur.com/zKLLWIr.png"/>
        </span>
      </template>

      <template slot="desktop-menu">
        <z-footer-list>
          <template slot="heading">Product</template>
          <z-footer-list-item>
            Autofix
          </z-footer-list-item>
          <z-footer-list-item>
            Code formatter
          </z-footer-list-item>
          <z-footer-list-item>
            Pricing
          </z-footer-list-item>
          <z-footer-list-item>
            Security
          </z-footer-list-item>
        </z-footer-list>

        <z-footer-list>
          <template slot="heading">Languages</template>
          <z-footer-list-item>
            For Python
          </z-footer-list-item>
          <z-footer-list-item>
            For Go
          </z-footer-list-item>
          <z-footer-list-item>
            For Ruby
          </z-footer-list-item>
          <z-footer-list-item>
            For JavaScript
          </z-footer-list-item>
        </z-footer-list>

        <z-footer-list>
          <template slot="heading">Resources</template>
          <z-footer-list-item>
            Documentation
          </z-footer-list-item>
          <z-footer-list-item>
            Blog
          </z-footer-list-item>
          <z-footer-list-item>
            Changelog
          </z-footer-list-item>
          <z-footer-list-item>
            Slack user group
          </z-footer-list-item>
        </z-footer-list>

        <z-footer-list>
          <template slot="heading">Company</template>
          <z-footer-list-item>
            About
          </z-footer-list-item>
          <z-footer-list-item>
            Customers
          </z-footer-list-item>
          <z-footer-list-item>
            Jobs
          </z-footer-list-item>
          <z-footer-list-item>
            Privacy Policy
          </z-footer-list-item>
          <z-footer-list-item>
            Terms of Service
          </z-footer-list-item>
          <z-footer-list-item>
            Press Enquiries
          </z-footer-list-item>
          <z-footer-list-item>
            Brand Assets
          </z-footer-list-item>
        </z-footer-list>

        <z-footer-list class="w-64">
          <template slot="heading">Social media</template>
          <z-footer-list-item hover-color="" :cursor-pointer="false">
            Follow us on social media to stay updated.
          </z-footer-list-item>
          <span class="flex mt-1">
            <z-footer-list-item class="mr-8">
              <z-icon icon="facebook"></z-icon>
            </z-footer-list-item>
            <z-footer-list-item class="mr-8">
              <z-icon icon="linkedin"></z-icon>
            </z-footer-list-item>
            <z-footer-list-item>
              <z-icon icon="mail"></z-icon>
            </z-footer-list-item>
          </span>
        </z-footer-list>
      </template>
      
      <template slot="desktop-cta">
        <img class="max-w-none" src="https://i.imgur.com/3wHn03E.png"/>
        <img class="max-w-none" src="https://i.imgur.com/3wHn03E.png"/>
      </template>
      
      <template slot="mobile-menu">
        <z-footer-list type="accordion">
          <template slot="heading">Product</template>
          <z-footer-list-item>
            Autofix
          </z-footer-list-item>
          <z-footer-list-item>
            Code formatter
          </z-footer-list-item>
          <z-footer-list-item>
            Pricing
          </z-footer-list-item>
          <z-footer-list-item>
            Security
          </z-footer-list-item>
        </z-footer-list>

        <z-footer-list type="accordion">
          <template slot="heading">Languages</template>
          <z-footer-list-item>
            For Python
          </z-footer-list-item>
          <z-footer-list-item>
            For Go
          </z-footer-list-item>
          <z-footer-list-item>
            For Ruby
          </z-footer-list-item>
          <z-footer-list-item>
            For JavaScript
          </z-footer-list-item>
        </z-footer-list>

        <z-footer-list type="accordion">
          <template slot="heading">Resources</template>
          <z-footer-list-item>
            Documentation
          </z-footer-list-item>
          <z-footer-list-item>
            Blog
          </z-footer-list-item>
          <z-footer-list-item>
            Changelog
          </z-footer-list-item>
          <z-footer-list-item>
            Slack user group
          </z-footer-list-item>
        </z-footer-list>

        <z-footer-list type="accordion">
          <template slot="heading">Company</template>
          <z-footer-list-item>
            About
          </z-footer-list-item>
          <z-footer-list-item>
            Customers
          </z-footer-list-item>
          <z-footer-list-item>
            Jobs
          </z-footer-list-item>
          <z-footer-list-item>
            Privacy Policy
          </z-footer-list-item>
          <z-footer-list-item>
            Terms of Service
          </z-footer-list-item>
          <z-footer-list-item>
            Press Enquiries
          </z-footer-list-item>
          <z-footer-list-item>
            Brand Assets
          </z-footer-list-item>
        </z-footer-list>

        <z-footer-list class="border-b border-slate py-6">
          <template slot="heading">
            Sign up for the newsletter
          </template>
          <span class="flex mt-1">
            <z-footer-list-item class="mr-8">
              <z-icon icon="facebook"></z-icon>
            </z-footer-list-item>
            <z-footer-list-item class="mr-8">
              <z-icon icon="linkedin"></z-icon>
            </z-footer-list-item>
            <z-footer-list-item>
              <z-icon icon="mail"></z-icon>
            </z-footer-list-item>
          </span>
        </z-footer-list>
      </template>
      
      <template slot="mobile-cta">
        <span class="flex justify-center py-8">
          <img class="w-20" src="https://i.imgur.com/3wHn03E.png"/>
          <img class="w-20" src="https://i.imgur.com/3wHn03E.png"/>
        </span>
      </template>

      <template slot="copyright">
        <span class="block text-center lg:text-left">
          Â© 2020, DeepSource Corp. All rights reserved.
        </span>
      </template>

      <template slot="additional-info">
        <span class="flex items-center mb-8 lg:mb-0">
          Backed by 
          <img class="ml-3 max-w-none" src="https://i.imgur.com/lnlRvjx.png"/>
        </span>
      </template>
    </z-footer>
  </div>
  `
})
