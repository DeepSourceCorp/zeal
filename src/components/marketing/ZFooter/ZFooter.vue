<template>
  <footer
    class="w-screen flex flex-col justify-between bg-transparent text-vanilla-100 border-t border-slate min-h-102 py-10"
  >
    <div
      class="lg:flex items-start w-screen lg:mx-auto"
      :class="[`${CONTAINERS[container].classes}`]"
    >
      <!-- Brand -->
      <div v-if="$slots.brand" :class="[CONTAINER_PADDING]">
        <slot name="brand"></slot>
      </div>

      <span class="hidden lg:flex items-start w-full">
        <div v-if="$slots['desktop-menu']" class="flex flex-1 items-start mx-4">
          <slot name="desktop-menu"></slot>
        </div>
        <div v-if="$slots['desktop-cta']" class="flex items-start" :class="[CONTAINER_PADDING]">
          <slot name="desktop-cta"></slot>
        </div>
      </span>

      <span class="block lg:hidden w-full">
        <div v-if="$slots['mobile-menu']" class="block">
          <slot name="mobile-menu"></slot>
        </div>
        <div v-if="$slots['mobile-cta']" :class="[CONTAINER_PADDING]">
          <slot name="mobile-cta"></slot>
        </div>
      </span>
    </div>
    <div
      class="flex flex-col-reverse lg:flex-row items-center justify-between w-screen lg:mx-auto text-vanilla-400 text-sm"
      :class="[`${CONTAINERS[container].classes}`, CONTAINER_PADDING]"
    >
      <div v-if="$slots.copyright">
        <slot name="copyright"></slot>
      </div>
      <div v-if="$slots['additional-info']">
        <slot name="additional-info"></slot>
      </div>
    </div>
  </footer>
</template>

<script lang="ts">
import Vue from 'vue'

const CONTAINER_PADDING = 'px-6'
const CONTAINERS = {
  sm: { text: 'sm', classes: 'lg:max-w-5xl' },
  md: { text: 'md', classes: 'lg:max-w-6xl' },
  lg: { text: 'lg', classes: 'lg:max-w-7xl' }
}

export default Vue.extend({
  components: {},
  name: 'ZFooter',
  props: {
    container: {
      type: String,
      default: CONTAINERS.md.text,
      validator: (size) => Object.keys(CONTAINERS).includes(size)
    }
  },
  data() {
    return {
      CONTAINERS,
      CONTAINER_PADDING
    }
  }
})
</script>
