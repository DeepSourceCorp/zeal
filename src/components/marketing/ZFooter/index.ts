import ZFooter from './ZFooter.vue'

export default ZFooter