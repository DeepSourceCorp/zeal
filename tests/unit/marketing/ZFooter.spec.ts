import Vue from 'vue'
import { createLocalVue, mount, Wrapper } from '@vue/test-utils'

import ZFooter from '../../../src/components/marketing/ZFooter'

describe('ZFooterList component', () => {
  let mountFunction: (options?: Record<string, unknown>) => Wrapper<Vue>
  let localVue: typeof Vue
  beforeEach(() => {
    localVue = createLocalVue()
    mountFunction = (options = {}) => {
      return mount(ZFooter, {
        localVue,
        ...options
      })
    }
  })
  it('renders a ZFooter when the component is mounted', () => {
    expect(mountFunction().html()).toMatchSnapshot()
  })
  it('renders a ZFooter with proper container size', () => {
    let mountedInstance = mountFunction({
      propsData: {
        container: 'md'
      }
    })
    expect(mountedInstance.html()).toMatchSnapshot()
    expect(mountedInstance.html()).toContain('lg:max-w-6xl')
  })
})
