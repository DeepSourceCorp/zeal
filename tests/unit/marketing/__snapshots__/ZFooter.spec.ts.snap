// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ZFooterList component renders a ZFooter when the component is mounted 1`] = `
<footer class="w-full flex flex-col justify-between bg-transparent text-vanilla-100 border-t border-slate-400 min-h-102 pb-10 pt-20">
  <div class="md:flex items-start w-full lg:mx-auto lg:max-w-6xl">
    <!----> <span class="hidden md:flex items-start w-full"><!----> <!----></span> <span class="block md:hidden w-full"><!----> <!----></span>
  </div>
  <div class="flex flex-col-reverse md:flex-row items-center justify-between w-full md:mx-auto text-vanilla-400 text-sm lg:max-w-6xl px-6 md:px-4 lg:px-6">
    <!---->
    <!---->
  </div>
</footer>
`;

exports[`ZFooterList component renders a ZFooter with proper container size 1`] = `
<footer class="w-full flex flex-col justify-between bg-transparent text-vanilla-100 border-t border-slate-400 min-h-102 pb-10 pt-20">
  <div class="md:flex items-start w-full lg:mx-auto lg:max-w-6xl">
    <!----> <span class="hidden md:flex items-start w-full"><!----> <!----></span> <span class="block md:hidden w-full"><!----> <!----></span>
  </div>
  <div class="flex flex-col-reverse md:flex-row items-center justify-between w-full md:mx-auto text-vanilla-400 text-sm lg:max-w-6xl px-6 md:px-4 lg:px-6">
    <!---->
    <!---->
  </div>
</footer>
`;
